dbms/src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -449,6 +449,24 @@ class AggregateFunctionSum final
             static_cast<ColumnVector<TResult> &>(to).getData().push_back(this->data(place).get());
     }
 
+    void batchInsertSameResultInto(ConstAggregateDataPtr __restrict place, IColumn & to, size_t num, Arena *)
+        const override
+    {
+        if constexpr (IsDecimal<TResult>)
+        {
+            // TODO refine it
+            auto & container = static_cast<ColumnDecimal<TResult> &>(to).getData();
+            container.reserve(container.size() + num);
+            for (size_t i = 0; i < num; ++i)
+                static_cast<ColumnDecimal<TResult> &>(to).getData().push_back(this->data(place).get(), result_scale);
```

Comment:
why not just `static_cast<ColumnDecimal<TResult> &>(to).getData().resize_fill(xxx)`?

Reply:
> why not just `static_cast<ColumnDecimal<TResult> &>(to).getData().resize_fill(xxx)`?

done